typer/main.py

```diff
@@ -610,7 +610,10 @@ def get_click_param(
             default_value = parameter_info.default
     elif param.default == Required or param.default == param.empty:
         required = True
-        parameter_info = ArgumentInfo()
+        if param.kind == inspect.Parameter.KEYWORD_ONLY:
```

Comment:
Where should I write the test?

Reply:
Ideally, there would be a tutorial file in `docs_src` demonstrating the behavior. And then a test for that file in `tests/test_tutorial/`. Then reference the file in the markdown documentation with a brief explanation.

Sometimes there are tricky edge cases that don't fit into tutorials. In that case, you can add some tests in the top level of the `tests/` directory.